x/kavadist/module.go

```diff
@@ -28,115 +31,135 @@ var (
+		AppModuleBasic: NewAppModuleBasic(),
 		keeper:         keeper,
-		supplyKeeper:   supplyKeeper,
+		accountKeeper:  accountKeeper,
 	}
 }
 
-// Name module name
-func (AppModule) Name() string {
-	return ModuleName
+// Name returns kavadist module's name.
+func (am AppModule) Name() string {
+	return am.AppModuleBasic.Name()
 }
 
-// RegisterInvariants register module invariants
-func (AppModule) RegisterInvariants(_ sdk.InvariantRegistry) {}
+// Route returns kavadist module's message route.
+func (am AppModule) Route() sdk.Route { return sdk.Route{} }
 
-// Route module message route name
-func (AppModule) Route() string {
-	return ModuleName
-}
+// QuerierRoute returns kavadist module's query routing key.
+func (AppModule) QuerierRoute() string { return types.QuerierRoute }
 
-// NewHandler module handler
-func (am AppModule) NewHandler() sdk.Handler {
-	return NewHandler(am.keeper)
+// LegacyQuerierHandler returns kavadist module's Querier.
+func (am AppModule) LegacyQuerierHandler(legacyQuerierCdc *codec.LegacyAmino) sdk.Querier {
+	return keeper.NewQuerier(am.keeper, legacyQuerierCdc)
 }
 
-// QuerierRoute module querier route name
-func (AppModule) QuerierRoute() string {
-	return ModuleName
+// RegisterServices registers a GRPC query service to respond to the
+// module-specific GRPC queries.
+func (am AppModule) RegisterServices(cfg module.Configurator) {
+	types.RegisterQueryServer(cfg.QueryServer(), am.keeper)
 }
 
-// NewQuerierHandler returns no sdk.Querier.
-func (am AppModule) NewQuerierHandler() sdk.Querier {
-	return keeper.NewQuerier(am.keeper)
-}
-
-// InitGenesis module init-genesis
-func (am AppModule) InitGenesis(ctx sdk.Context, data json.RawMessage) []abci.ValidatorUpdate {
-	var genesisState GenesisState
-	ModuleCdc.MustUnmarshalJSON(data, &genesisState)
-	InitGenesis(ctx, am.keeper, am.supplyKeeper, genesisState)
+// RegisterInvariants registers kavadist module's invariants.
+func (am AppModule) RegisterInvariants(_ sdk.InvariantRegistry) {}
 
+// InitGenesis performs kavadist module's genesis initialization It returns
+// no validator updates.
+func (am AppModule) InitGenesis(ctx sdk.Context, cdc codec.JSONCodec, gs json.RawMessage) []abci.ValidatorUpdate {
+	var genState types.GenesisState
+	cdc.MustUnmarshalJSON(gs, &genState)
+	InitGenesis(ctx, am.keeper, am.accountKeeper, &genState)
 	return []abci.ValidatorUpdate{}
 }
 
-// ExportGenesis module export genesis
-func (am AppModule) ExportGenesis(ctx sdk.Context) json.RawMessage {
-	gs := ExportGenesis(ctx, am.keeper)
-	return ModuleCdc.MustMarshalJSON(gs)
+// ExportGenesis returns kavadist module's exported genesis state as raw JSON bytes.
+func (am AppModule) ExportGenesis(ctx sdk.Context, cdc codec.JSONCodec) json.RawMessage {
+	genState := ExportGenesis(ctx, am.keeper)
+	return cdc.MustMarshalJSON(genState)
 }
 
-// BeginBlock module begin-block
+// ConsensusVersion implements ConsensusVersion.
+func (AppModule) ConsensusVersion() uint64 { return 2 }
```

Comment:
Just a minor thing, should return 1 instead of 2:
https://docs.cosmos.network/master/architecture/adr-041-in-place-store-migrations.html#module-consensusversion

```suggestion
func (AppModule) ConsensusVersion() uint64 { return 1 }
```

Reply:
Oops too late now but we can update it later

Reply:
opps, yeah i'll need to update this. will create a PR

Reply:
https://github.com/Kava-Labs/kava/pull/1039